zrml/prediction-markets/src/lib.rs

```diff
@@ -99,13 +93,9 @@ mod pallet {
             // TODO(#618): Not implemented for Rikiddo!
             T::DestroyOrigin::ensure_origin(origin)?;
 
-            let mut total_accounts = 0usize;
-            let mut share_accounts = 0usize;
             let market = T::MarketCommons::market(&market_id)?;
             ensure!(market.scoring_rule == ScoringRule::CPMM, Error::<T>::InvalidScoringRule);
             let market_status = market.status;
-            let outcome_assets = Self::outcome_assets(market_id, &market);
-            let outcome_assets_amount = outcome_assets.len();
```

Comment:
The market will be deleted completely out of storage later and we won't know which outcome assets to delete in the future. I think this is fine though, since we can deduct which which markets got deleted (we got incremental market ids and some won't return a market -> delete markets) and then iterate through all accounts in a (potential multi-block) migration and delete outcome assets where the market id belongs to one of the deleted markets.

Reply:
That sounds like a reasonable approach!